src/style/text_style.rs

```diff
@@ -12,20 +12,18 @@ pub trait TextStyle {
     /// Color type.
     type Color: PixelColor;
 
-    /// Render a text object using this style.
-    fn render_text<D>(&self, text: &Text<'_>, target: &mut D) -> Result<(), D::Error>
+    /// Renders a single line of text using this style.
+    ///
+    /// Returns the offset from the current position to the start of the next line.
```

Comment:
I think this doc could use some `# Implementation notes`;

* For the first line, what's the starting `position`?
* For subsequent lines, isn't the position just `position + Point::new(0, <line_y_height>)`? It would be good to clarify this behaviour.
* Does the `position` within a line matter? E.g. should it be the top left corner of the line, the baseline, etc?

Edit: After writing the other comment, should this be renamed to `render_first_line` (or something consistent with the other bounding box method name)?

Reply:
There will be some changes to the trait in future PR to add additional formatting options, like horizontal and vertical alignment.

> * For the first line, what's the starting `position`?

It only renders a single line. The meaning of `position` will change in the future, so I decided not to docment it right now.

>* For subsequent lines, isn't the position just `position + Point::new(0, <line_y_height>)`?

Yes, it should. The method returns the offset, because `Text` doesn't known the line height.

> * Does the `position` within a line matter? E.g. should it be the top left corner of the line, the baseline, etc?

This will be added later.

> After writing the other comment, should this be renamed to `render_first_line`

It's used for all lines, so `render_first_line` wouldn't be a good name. But I'll rename it to `render_line`.